src/packages/StorageFileApi.ts

```diff
@@ -80,22 +82,39 @@ export default class StorageFileApi {
     try {
       let body
       const options = { ...DEFAULT_FILE_OPTIONS, ...fileOptions }
-      const headers: Record<string, string> = {
+      let headers: Record<string, string> = {
         ...this.headers,
         ...(method === 'POST' && { 'x-upsert': String(options.upsert as boolean) }),
       }
 
+      const userMetadata = options.metadata
+
       if (typeof Blob !== 'undefined' && fileBody instanceof Blob) {
         body = new FormData()
         body.append('cacheControl', options.cacheControl as string)
         body.append('', fileBody)
+
+        if (userMetadata) {
+          body.append('userMetadata', this.encodeMetadata(userMetadata))
```

Comment:
i thought we decided to rename this to `customMetadata` in the backend?

Reply:
`user_metadata` is fine on the backend, but yes, the input here should be `metadata`